rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleRadioOperationCustom.java

```diff
@@ -0,0 +1,37 @@
+package com.polidea.rxandroidble;
+
+import android.bluetooth.BluetoothGatt;
+
+import com.polidea.rxandroidble.internal.connection.RxBleGattCallback;
+
+import rx.Observable;
+import rx.Observer;
+import rx.Scheduler;
+import rx.android.schedulers.AndroidSchedulers;
+
+/**
+ * Represents a custom operation that will be enqueue for future execution within the client instance.
+ */
+public interface RxBleRadioOperationCustom<T> {
+
+    /**
+     * Execute the custom operation. This method is called once the {@link com.polidea.rxandroidble.internal.RxBleRadio}
+     * is determining that this operation should be the next to be run.
+     * <p>
+     * The method receives everything needed to access the low-level Android BLE API objects.
+     * <p></p>
+     * As the implementer, you access {@link BluetoothGatt} and {@link RxBleGattCallback} to implement
+     * a custom operation and you use the {@link Observer<T>} {@code observerProxy} object to forward
+     * events to {@link rx.Observable<T>} return by the {@link RxBleConnection#queue(RxBleRadioOperationCustom)}
+     * operation. You <b>must</b> call {@link Observer#onCompleted() or {@link Observer#onError(Throwable)} otherwise
+     * the {@link rx.Observable<T>} will never really terminate.
```

Comment:
will never really terminate and the operations queue will not continue to process next events.